pkg/golinters/depguard.go

```diff
@@ -0,0 +1,58 @@
+package golinters
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	depguardAPI "github.com/OpenPeeDeeP/depguard"
+	"github.com/golangci/golangci-lint/pkg/result"
+)
+
+type Depguard struct{}
+
+func (Depguard) Name() string {
+	return "depguard"
+}
+
+func (Depguard) Desc() string {
+	return "Go linter that checks if package imports are in a list of acceptable packages"
+}
+
+func (d Depguard) Run(ctx context.Context, lintCtx *Context) ([]result.Issue, error) {
+	dg := &depguardAPI.Depguard{
+		Packages:      lintCtx.Settings().Depguard.Packages,
+		IncludeGoRoot: lintCtx.Settings().Depguard.IncludeGoRoot,
+	}
+	listType := lintCtx.Settings().Depguard.ListType
+	if listType == "" {
+		listType = "whitelist"
+	}
+
+	var found bool
+	dg.ListType, found = depguardAPI.StringToListType[strings.ToLower(listType)]
+	if !found {
+		return nil, fmt.Errorf("unsure what list type %s is", listType)
+	}
+
+	issues, err := dg.Run(lintCtx.LoaderConfig, lintCtx.Program)
+	if err != nil {
+		return nil, err
+	}
+	if len(issues) == 0 {
+		return nil, nil
+	}
+	msgSuffix := "is in the blacklist"
+	if dg.ListType == depguardAPI.LTWhitelist {
+		msgSuffix = "is not in the whitelist"
+	}
+	res := make([]result.Issue, 0, len(issues))
+	for _, i := range issues {
+		res = append(res, result.Issue{
+			Pos:        i.Position,
+			Text:       fmt.Sprintf("%s %s", i.PackageName, msgSuffix),
```

Comment:
use `formatCode` for `i.PackageName` please: it will be nice-formatted in GitHub comments